js/sdks/packages/cli/src/docker-compose/lunasec-stack.ts

```diff
@@ -569,7 +632,10 @@ export class LunaSecStackDockerCompose {
 
     writeFileSync(composePath, dockerCompose);
 
-    const dockerEnv = this.getDockerEnv();
+    // TODO (freeqaz) Create a function for every environment that we support and
+    // have it merge in all of them using something like Ramda.pipe()
+    // As long as every function is referentially transparent, it's a clean abstraction.
+    const dockerEnv = generateNginxEnvConfig(this.env, this.getDockerEnv());
```

Comment:
this feels like a pretty confusing pattern since env values are going to be overwritten in a possibly confusing way. since we need to overhaul how the config is being created, I'm going to say this is ok for now since it works. creating CDK like constructs for each service will let us customize how each service presents itself to other services depending on what environment it is in:
```
tokenizerBackend = new TokenizerBackend(env);
applicationBackend = new ApplicationBackend(env, {
  tokenizerBackend,
});

class ApplicationBackend extends DockerComposeService {
  envName = 'APPLICATION_BACKEND';
  hostname = 'application-back-end';

  constructor(env, deps) {
    super(env, deps);
  }
}

class DockerComposeService {
  envVars: Record<string, string> = {}
  hostname: string | null = null;
  port: number | null = null;
  protocol: 'http' | 'https' | null = null;

  constructor(env, deps) {
    this.envVars = deps.reduce((dep, envVars) => {
      return {
        ...envVars,
        [dep
      }
    }, {});
  }

  getIntraNetworkName() {
    return `INTRA_NET_${this.envName}`;
  }

  getInterNetworkName() {
    return `INTER_NET_${this.envName}`;
  }

  getIntraNetworkValue() {
    if (this.hostname === null) {
      raise new Error('hostname is null');
    }
    ...
    return `${this.protocol}://${this.hostname}:${this.port}`;
  }

  getInterNetworkValue() {
    return `${this.protocol}://localhost:${this.port}`;
  }
}
```